### PR TITLE
Bug 1193493 - Test video with limited bandwidth.

### PR DESCRIPTION
**Pull request by sydvicious**
Test that video playback works when playback is limited. Note that this set of tests sets the bandwidth and then runs the videos. Another issue and set of tests will be developed to vary the bandwidth.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -141,6 +141,24 @@ Crash address: 0x104616900
 ...
 ```
 
+### Setting up for network shaping tests (browsermobproxy)
+
+Setup
```
Comment (mjzffr) [inline README.md]: Remove extra heading.

File: README.md [hunk 2]
```diff
@@ -141,6 +141,24 @@ Crash address: 0x104616900
...
+-----
+
+1. Download the browsermob proxy zip file from http://bmp.lightbody.net/. The most current version as of this writing is browsermob-proxy-2.1.0-beta-2-bin.zip.
+2. Unpack the .zip file.
+3. Import the certificate into your Firefox profile. Select Preferences->Advanced->Certificates->View Certificates->Import... Navigate to <browsermob>/ssl-support and select cybervilliansCA.cer. Select all of the checkboxes.
```
Comment (mjzffr) [inline README.md]: `<browsermob>` doesn't show up in the generated HTML. Maybe replace with `\<browsermob\>`.

**Comment by mjzffr** [comment]
There are conflicts because I force-pushed a change this morning. They will go away once you rebase your branch only the latest master and git push -f to origin.

File: README.md [hunk 3]
```diff
@@ -141,6 +141,24 @@ Crash address: 0x104616900
...
+4. Tell marionette where browsermobproxy is and what port to start it on. Add the following command-line parameters to your firefox-media-tests command line:
+
+<pre><code>
+--browsermob-script <browsermob>/bin/browsermob-proxy --browsermob-port 999
+</code></pre>
```
Comment (mjzffr) [inline README.md:154]: ...and which profile to use

File: firefox_media_tests/playback/test_playback_limiting_bandwidth.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette import BrowserMobProxyTestCaseMixin
+
+from media_utils.video_puppeteer import VideoPuppeteer
+from testcase import MediaTestCase
```
Comment (mjzffr) [inline firefox_media_tests/playback/test_playback_limiting_bandwidth.py]: Should be `from media_test_harness.testcase import MediaTestCase`

**Comment by mjzffr** [comment]
Trying to reproduce your instructions. Command:

```
firefox-media-tests --binary $FF_NIGHTLY_PATH --profile ~/dev/temp/firefox-profiles/tyel4j1n.browsermob firefox_media_tests/playback/limiting_bandwidth.ini --browsermob-script ~/dev/tools/browsermob-proxy-2.1.0-beta-2/bin/browsermob-proxy --browsermob-port 999
```

Browser loads, but no page is loaded. Test logs stuck at: `0:01.62 TEST_START: MainThread test_playback_limiting_bandwidth.py TestPlaybackLimitingBandwidth.test_playback_limiting_bandwidth_1000`

Seems to be stuck at:   `File "/Users/mozilla/dev/firefox-media-tests/firefox_media_tests/playback/test_playback_limiting_bandwidth.py", line 22, in setUp
    self.proxy = self.create_browsermob_proxy()`

Eventually I get an error: raise Exception("Can't connect to Browsermob-Proxy")

I have verified that the `cybervillains` certificate is being used by the Firefox binary being tested; have tried with different ports, e.g. 50000 in the ephemeral range.

**Comment by sydvicious** [comment]
Can you launch browsermob proxy in a shell using the script? You may have to mess around with Java. It was a pain to get it to work for me. I downloaded the jdk, and then I had to remake the /usr/bin/java symlink to point to it.

**Comment by mjzffr** [comment]
I'll try in a VM and get back to you. Maybe worthwhile to add a note about this issue in the readme.

**Comment by sydvicious** [comment]
I am torn. I really don't want to do browsermobproxy support. You should be able to figure out how to launch it using the script yourself. But I can put the workaround in if we really really need it....

**Comment by mjzffr** [comment]
Yeah, all I had in mind was a very brief note that, hey, if the bandwidth limiting tests don't run, it could be an issue with browsermob+java.

**Comment by sydvicious** [comment]
I think a better idea is to make sure that you can launch browsermobproxy yourself before running the tests.

**Comment by mjzffr** [comment]
Agreed.

**Comment by sydvicious** [comment]
@mjzffr , feedback incorporated.

**Pull request merged**